app/models/user.rb

```diff
@@ -21,4 +24,31 @@ def likes?(post)
     like = Like.where('post_id = ? and user_id = ?', post.id, id)
     like.count.positive?
   end
+
+  # returns an array of friend who are friends with a particular user.
+  def friends
+    friendships_array = friendships.map { |friendship| friendship.friend if friendship.status }
+    inverse_friendships_array = inverse_friendships.map { |friendship| friendship.user if friendship.status }
+    friendships_array << inverse_friendships_array
+    friends_array.compact
+  end
+
+  # returns
+  def pending_friends
+    friendships.map { |f| f unless f.status }.compact
+  end
+
+  def friend_requests
+    inverse_friendships.map { |f| f unless f.status }.compact
+  end
+
+  def confirm_friend(user)
+    friendship = inverse_friendships.find { |f| f.user == user }
```

Comment:
This isn't really doing anything. This is an association you could query by doing:

```
irb(main):031:0> u.inverse_friendships.find_by(user_id: 2)
  Friendship Load (0.6ms)  SELECT  "friendships".* FROM "friendships" WHERE "friendships"."friend_id" = $1 AND "friendships"."user_id" = $2 LIMIT $3  [["friend_id", 1], ["user_id", 2], ["LIMIT", 1]]
=> #<Friendship id: 8, user_id: 2, friend_id: 1, status: false, created_at: "2019-12-27 23:45:54", updated_at: "2019-12-27 23:45:54">
```

You have the user an argument, and all you really need is its id number to make the query, grab the record, and switch status to `true`, but since this method isn't working, that's why we get:

```
irb(main):033:0> Friendship.where(status: true)
  Friendship Load (0.6ms)  SELECT  "friendships".* FROM "friendships" WHERE "friendships"."status" = $1 LIMIT $2  [["status", "t"], ["LIMIT", 11]]
=> #<ActiveRecord::Relation []>
```

It should contain the recently created friendship record, right?

[`REQUIRED`]